### PR TITLE
pkg/tinycbor: Initial import

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

Initial import of the [tinycbor](https://github.com/intel/tinycbor) package.

From an initial impression it is less memory hungry than the cn-cbor package.

Test is based on the cn-cbor test.

### Issues/PRs references

depends on #8793 due to a header naming conflict

### CODE REVIEW

**Comment by bergzand** [comment]
I've added some code to the makefile to effectively use `tinycbor/cbor.h" in the code. The sed rules are to adapt the current cbor code. I can create a patch for it instead if preferred.

With this the conflict between the current cbor module and this package is solved.

File: pkg/tinycbor/Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+PKG_NAME=tinycbor
+PKG_URL=https://github.com/intel/tinycbor
+PKG_VERSION=v0.5.1
+PKG_LICENSE=MIT
+
+.PHONY: all
+
+all: git-download
+	sed -i 's/"cbor.h"/"tinycbor\/cbor.h"/' $(PKG_BUILDDIR)/src/*.h
```
Comment (kaspar030) [inline pkg/tinycbor/Makefile]: ```*.[hc]```?

File: tests/pkg_tinycbor/main.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ *
+ * @}
+ */
+
+#include "tinycbor/cbor.h"
+
+#include "fmt.h"
+#include "embUnit.h"
+#include "embUnit/embUnit.h"
+#include <string.h>
```
Comment (kaspar030) [inline tests/pkg_tinycbor/main.c]: please put the standard header on top

File: tests/pkg_tinycbor/main.c [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+static void test_parse(void)
+{
+    static size_t idx;
+    char *tests[] = {
```
Comment (kaspar030) [inline tests/pkg_tinycbor/main.c]: would it save stack here to make these file-scope and const?

File: tests/pkg_tinycbor/main.c [hunk 4]
```diff
@@ -0,0 +1,106 @@
+/*
```
Comment (kaspar030) [inline tests/pkg_tinycbor/main.c]: This is laid out like our unittests, right?

**Comment by bergzand** [comment]
Noticed that `Makefile.tinycbor` was not behaving as expected. Fixed that in the last fixup commit.

**Comment by cladmi** [comment]
I have an idea to does not require changing the package.

As the `cbor` package is deprecated, I think it would be better to remove `cbor.h` from the standard `sys/include` like this ?

``` diff
diff --git a/sys/Makefile.include b/sys/Makefile.include
index 24b8c72..aac2784 100644
--- a/sys/Makefile.include
+++ b/sys/Makefile.include
@@ -6,6 +6,10 @@ ifneq (,$(filter crypto,$(USEMODULE)))
   USEMODULE_INCLUDES += $(RIOTBASE)/include/crypto
 endif
 
+ifneq (,$(filter cbor,$(USEMODULE)))
+  USEMODULE_INCLUDES += $(RIOTBASE)/sys/cbor/include
+endif
+
 ifneq (,$(filter fib,$(USEMODULE)))
   USEMODULE_INCLUDES += $(RIOTBASE)/sys/posix/include
 endif
diff --git a/sys/include/cbor.h b/sys/cbor/include/cbor.h
similarity index 100%
rename from sys/include/cbor.h
rename to sys/cbor/include/cbor.h
```

**Comment by bergzand** [comment]
#9055 is merged, so I removed the sed rules.

**Comment by bergzand** [comment]
No longer waiting on a PR, ready for review/merge

File: tests/pkg_tinycbor/tests/01-run.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import sys
+
+
+def testfunc(child):
+    child.expect(r"OK \(\d+ tests\)")
```
Comment (kYc0o) [inline tests/pkg_tinycbor/tests/01-run.py]: Can you please add the amount of tests that are actually expected to be OK? Is it possible?

Comment (bergzand) [reply tests/pkg_tinycbor/tests/01-run.py]: Done

**Comment by bergzand** [comment]
I've enabled more strict checking for the cbor streams

**Comment by bergzand** [comment]
I have a few changes open upstream to fix the compilation issues for the 8 and 16 bit platforms. For now I'll disable those architectures for this package, but hopefully we can enabled them again with the next release of tinycbor

**Comment by bergzand** [comment]
@kYc0o Murdock agrees now. Can I squash again?

@kaspar030 Are your issues solved here?

**Comment by kaspar030** [comment]
(please squash!)

**Comment by bergzand** [comment]
Squashed!

**Comment by kYc0o** [comment]
Good, fro now it's ok that it works only for 32bit platforms, let's wait for the next release, hopefully the upstream changes are merged by then.

Thanks!

**Pull request merged**